### PR TITLE
Fix/eth_getBlockByNumber and eth_getUncle... nullable fields

### PR DESCRIPTION
**Pull request by meowsbits**
Rel #334

This is working toward implementing and testing `eth_getBlockByNumber` for `pending` blocks achieving complete parity with ethereum/go-ethereum (... and with reference to the docs, which seem to be occasionally a source of noise rather than truth).

One question remaining concerns the `totalDifficulty` field. Currently (as tested) both core-geth and go-ethereum return `null` for the `pending` block. This field is _not present_ in the docs (cited at above ref'd issue). Logic at the API level determines the existence of this field conditionally on an `inclTxes` boolean. I'm remembering wrangling with this a while ago, but need to do some more digging to get the details again.

### CODE REVIEW

**Comment by meowsbits** [comment]
Ref https://github.com/etclabscore/ethereum.go-ethereum/pull/11/files. This branch shows that `ethclient.GetHeader(-1)`, which should return the `pending` block, in fact does not.  This is a bug in the `ethclient` implementation (although with roots potentially at `types.Header` JSON Unmarshal). Note that this branch and the demonstrated bug is at ethereum/go-ethereum. 

Though related, _this_ issue is first concerned with the RPC service (ie server) providing valid information (rather than the Go `ethclient` implementation of a client consuming this data properly).

File: internal/ethapi/api.go [hunk 1]
```diff
@@ -1198,12 +1198,7 @@ func NewRPCMarshalHeaderTFromHeader(header *types.Header) *RPCMarshalHeaderT {
 // rpcMarshalHeaderTSetTotalDifficulty sets the total difficulty field for RPC response headers.
 // If the hash is unavailable (ie in Pending state), the value will be 0.
 func (s *PublicBlockChainAPI) rpcMarshalHeaderTSetTotalDifficulty(ctx context.Context, header *RPCMarshalHeaderT) {
-	hash := header.Hash
-	if hash == nil {
-		c := common.Hash{}
-		hash = &c
-	}
-	header.TotalDifficulty = (*hexutil.Big)(s.b.GetTd(ctx, *hash))
+	header.TotalDifficulty = (*hexutil.Big)(s.b.GetTd(ctx, *header.Hash))
```
Comment (meowsbits) [inline internal/ethapi/api.go:1201]: This now matches the ethereum/go-ethereum implementation.
> https://github.com/ethereum/go-ethereum/blob/master/internal/ethapi/api.go#L1187-L1199

**Comment by meowsbits** [comment]
Re: `totalDifficulty`.
- https://github.com/ethereum/go-ethereum/pull/20460
- https://github.com/ethereum/go-ethereum/issues/19024#issuecomment-505358363

I've set it to behave equivalently with ethereum/go-ethereum. This means that it will be included (potentially `null`... under certain conditions...). This might spin off into it's own issue.

**Comment by meowsbits** [comment]
This commit https://github.com/etclabscore/core-geth/pull/341/commits/67ca5fcf0398beba097b5ba27517d8079ad7ea58 fixes this issue: https://github.com/etclabscore/core-geth/issues/344

**Comment by meowsbits** [comment]
There was a `panic` at the CI. https://travis-ci.org/github/etclabscore/core-geth/jobs/763301644

At first glance it seems unrelated to this change set, but has now happened twice in a row (after 1 re-run). Will look into it tomorrow.

**Review by ziogaschr — APPROVED** [review]
LGTM

**Pull request merged**